src/olympia/accounts/views.py

```diff
@@ -487,3 +490,56 @@ def post(self, request):
             'fxa_id': user.fxa_id,
             'session_cookie': cookie,
         }, status=201)
+
+
+class AccountNotificationViewSet(ListModelMixin, GenericViewSet):
+    """Returns account notifications.  If not already set by the user defaults
+    will be returned."""
+
+    permission_classes = [IsAuthenticated]
+    # We're pushing the primary permission checking to AccountViewSet for ease.
+    account_permission_classes = [
+        AnyOf(AllowSelf, GroupPermission(amo.permissions.USERS_EDIT))]
+    serializer_class = UserNotificationSerializer
+    paginator = None
+
+    def get_account_viewset(self):
+        if not hasattr(self, 'account_viewset'):
+            self.account_viewset = AccountViewSet(
+                request=self.request,
+                permission_classes=self.account_permission_classes,
+                kwargs={'pk': self.kwargs['user_pk']})
+        return self.account_viewset
+
+    def _get_default_object(self, notification):
+        return UserNotification(
+            user=self.get_account_viewset().get_object(),
+            notification_id=notification.id,
+            enabled=notification.default_checked)
+
+    def get_queryset(self):
+        queryset = UserNotification.objects.filter(
+            user=self.get_account_viewset().get_object())
+        # Put it into a dict so we can easily check for existence.
+        set_notifications = {
+            user_nfn.notification.short: user_nfn for user_nfn in queryset}
+        out = []
+        for notification in NOTIFICATIONS:
+            out.append(set_notifications.get(
+                notification.short,  # It's been set by the user.
+                self._get_default_object(notification)))  # Otherwise, default.
+        return out
+
+    @list_route(methods=['patch'])
```

Comment:
The `method` is the only thing I'm not sure about. Since it's a dedicated endpoint, I would have expected a `post` instead. Am I the only one ?